lib/fileListHeaders.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @copyright Copyright (c) 2023 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { Folder } from './files/folder'
+import logger from './utils/logger'
+
+export interface HeaderData {
+	/** Unique ID */
+	id: string
+	/** Order */
+	order: number
+	/** Condition wether this header is shown or not */
+	enabled?: (folder: Folder, view) => boolean
```

Comment:
Should the view be typed somehow?

Reply:
I agree we should have the view types (if possible)

Reply:
Yes, we haven't migrated it from server yet! Was maing sure no changes were needed (I had to adjust quite a lot at the beginning)